src/System.IO.FileSystem.Watcher.Polling/System/IO/PollingFileSystemWatcher.cs

```diff
@@ -14,82 +11,73 @@
+    /// In scenarios where events cannot be missed, PollingFileSystemWatcher should be used.
     /// Note: When a watched file is renamed, one or two notifications will be made.
-    /// Note: When no changes are detected, PollingWatcher will not allocate memory on the GC heap.
+    /// Note: When no changes are detected, PollingFileSystemWatcher will not allocate memory on the GC heap.
     /// </remarks>
-    public class PollingWatcher : IDisposable
+    public class PollingFileSystemWatcher : IDisposable
     {
-        public TraceSource Tracing { get; private set; }
-        Stopwatch _stopwatch = new Stopwatch();
         Timer _timer;
-        int _pollingIntervalInMilliseconds;
-        bool _includeSubdirectories;
-
-        List<string> _extensionsToWatch;
-        string _directory;
         PathToFileStateHashtable _state; // stores state of the directory
         byte _version; // this is used to keep track of removals. // TODO: describe the algorithm
-        static string[] SpecialFiles = new[] { "", ".", ".." };
 
         /// <summary>
         /// Creates an instance of a watcher
         /// </summary>
         /// <param name="directory">The directory to watch. It does not support UNC paths (yet).</param>
-        /// <param name="includeSubdirectories">A bool controlling whether or not subdirectories will be watched too</param>
-        /// <param name="pollingIntervalInMilliseconds">Polling interval</param>
-        public PollingWatcher(string directory, bool includeSubdirectories = false, int pollingIntervalInMilliseconds = 1000)
+        /// <param name="filter">The type of files to watch. For example, "*.txt" watches for changes to all text files.</param>
+        public PollingFileSystemWatcher(string directory, string filter = "*.*")
         {
-            if (!Directory.Exists(directory))
+            if (!IO.Directory.Exists(directory))
                 throw new ArgumentException("Directory not found.", nameof(directory));
 
-            Tracing = new TraceSource("PollingWatcher");
-            _state = new PathToFileStateHashtable(Tracing);
-            _pollingIntervalInMilliseconds = pollingIntervalInMilliseconds;
-            _includeSubdirectories = includeSubdirectories;
-            _directory = directory;
+            _state = new PathToFileStateHashtable();
+            Directory = directory;
+            Filter = filter;
         }
 
         /// <summary>
-        /// Extensions to watch
+        /// Creates an instance of a watcher
         /// </summary>
-        /// <param name="extension">for example "txt", "doc", etc., i.e don't use wildcards</param>
-        /// <remarks>
-        /// By default all extensions are watched. Once this method is called, only extensions in added are watched.
-        /// </remarks>
-        public void AddExtension(string extension)
+        /// <param name="directory">The directory to watch. It does not support UNC paths (yet).</param>
```

Comment:
Why doesn't it support UNCs?

Reply:
Actually, that message is preexisting. Not sure why, though I never tested it. I will.

Reply:
Works fine. :)